### PR TITLE
Update Weights & Biases integration + Add Inference logging

### PR DESCRIPTION
**Pull request by soumik12345**
# Changes made in this PR

## Log inference results to Weights & Biases

Fixes https://github.com/Deci-AI/super-gradients/issues/953

Optionally log inference results for object detection in [Weights & Biases](https://wandb.ai/site) dashboard using [bounding-box image overlays](https://docs.wandb.ai/guides/track/log/media#image-overlays).

This PR adds an optional parameter `visualize_on_wandb` to `ImagesDetectionPrediction.show()` which when set to `True` would log the predictions using the [bounding-box image overlays](https://docs.wandb.ai/guides/track/log/media#image-overlays) in the [Weights & Biases](https://wandb.ai/site) dashboard of the user.

Here's a sample code:

```python
from super_gradients.training import models
from super_gradients.common.object_names import Models

import wandb

# Initialize Weights & Biases run
wandb.init(project="yolo-nas", entity="geekyrakshit")

net = models.get(Models.YOLO_NAS_S, pretrained_weights="coco")
prediction = net.predict([
    "https://cdn.autonomous.ai/static/upload/images/common/upload/20210919/5-Desk-Setup-with-a-Laptop--Monitor-Ideas-for-2021_10f89201411b.jpg",
    "https://p.imgci.com/db/PICTURES/CMS/352000/352019.jpg",
    "https://p.imgci.com/db/PICTURES/CMS/352200/352298.jpg",
    "https://www.encirclephotos.com/wp-content/uploads/Laos-Ban-Pak-Ou-Man-Riding-Asian-Elephant-1440x961.jpg"
])
prediction.show(visualize_on_wandb=True)
```

This would log the images to be visualized in a media panel in a Weights & Biases dashboard.

👇 Here's how the results would look
[](https://wandb.ai/geekyrakshit/yolo-nas-integration/runs/lz9mjhwy)

Here's a video of how to interact with the bounding-box overlay interface on Weights & Biases 👇

https://github.com/Deci-AI/super-gradients/assets/19887676/fb027d35-adc1-4e93-9d5a-2ea7b32c50f7

## Optionally decouple `wandb.init()` from `WandBSGLogger`

Fixes https://github.com/Deci-AI/super-gradients/issues/980

`WandBSGLogger` will now call `wandb.init()` only if it has not been called earlier.

👇 Here's a sample training code that works with the changes made in this PR
[](https://colab.research.google.com/drive/1-HrD-iUedZCHOnfT4topmASE_0RcC5RQ?usp=sharing)

👇 Here's a sample Weights & Biases Run
[](https://wandb.ai/geekyrakshit/yolo-nas-integration/runs/wkuetqdq)

### CODE REVIEW

**Comment by BloodAxe** [comment]
Hi! First of all, thanks for your PR. We really appreciate external contributions. 

However, there are a few things about this PR that prohibits it from being approved.

First, all commits in PR must be a [signed commits](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits). This wasn't be reflected in CONTRIBUTING.md and will be changed in future. But that's an internal company policy and we can't go beyond it.

Second issue is that this PR violate SRP responsibility of .show() method. It was specifically designed for plotting detection results in Colab environment using matplotlib. Adding W&B support to this method is not in alignment with our design ideas. I don't see any reason why W&B logging cannot be made a simple function `log_detection_results_to_wandb`:

```
from super_gradients.wandb import log_detection_results_to_wandb

prediction = net.predict(...)
log_detection_results_to_wandb(prediction)
```

To me, this version looks way more cleaner and we would be happy to have it contributed to our repo. If you're willing to proceed with this version please ensure you have set-up signed commits and re-open a new PR for this change.

Thanks!
